### ISSUE TITLE
Error while calling cost-estimation API

### ISSUE THREAD

**Issue by agatsoh** [raiden-network/raiden-wizard#231]
While calling cost-estimation API get this error
```
ERROR:tornado.application:Uncaught exception POST /api/cost-estimation/config-0x7233c7fD40A1b01C1579d6b1120Fdea38853af62-mainnet.toml (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:1994', method='POST', uri='/api/cost-estimation/config-0x7233c7fD40A1b01C1579d6b1120Fdea38853af62-mainnet.toml', version='HTTP/1.1', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/tornado/web.py", line 1697, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/devaki/raidenforked/raiden-installer/raiden_installer/web.py", line 708, in post
    exchange_costs = exchange.calculate_transaction_costs(token_amount, account)
  File "/home/devaki/raidenforked/raiden-installer/raiden_installer/token_exchange.py", line 59, in calculate_transaction_costs
    return self._calculate_transaction_costs(token_amount, account)
  File "/home/devaki/raidenforked/raiden-installer/raiden_installer/token_exchange.py", line 149, in _calculate_transaction_costs
    block = self.w3.eth.getBlock(self.w3.eth.blockNumber)
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/eth.py", line 217, in getBlock
    [block_identifier, full_transactions],
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/manager.py", line 149, in request_blocking
    response = self._make_request(method, params)
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/manager.py", line 128, in _make_request
    return request_func(method, params)
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/signing.py", line 150, in middleware
    return make_request(method, params)
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/cache.py", line 134, in middleware
    return make_request(method, params)
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/formatting.py", line 75, in apply_formatters
    response = make_request(method, params)
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/gas_price_strategy.py", line 34, in middleware
    return make_request(method, params)
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/formatting.py", line 75, in apply_formatters
    response = make_request(method, params)
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/attrdict.py", line 33, in middleware
    response = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/formatting.py", line 73, in apply_formatters
    response = make_request(method, formatted_params)
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/normalize_errors.py", line 25, in middleware
    result = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/formatting.py", line 75, in apply_formatters
    response = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/formatting.py", line 75, in apply_formatters
    response = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/web3/middleware/formatting.py", line 82, in apply_formatters
    formatter(response["result"]),
  File "cytoolz/functoolz.pyx", line 503, in cytoolz.functoolz.Compose.__call__
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/eth_utils/functional.py", line 45, in inner
    return callback(fn(*args, **kwargs))
  File "/home/devaki/Envs/installer/lib/python3.7/site-packages/eth_utils/applicators.py", line 119, in apply_key_map
    set(value.keys())
AttributeError: 'NoneType' object has no attribute 'keys'
ERROR:tornado.access:500 POST /api/cost-estimation/config-0x7233c7fD40A1b01C1579d6b1120Fdea38853af62-mainnet.toml (127.0.0.1) 19496.09ms
```

**Comment by fredo** [comment]
does this error still happens with the current amster for you @agatsoh ?

**Comment by agatsoh** [comment]
I will close the issue if it doesnt occur

**Comment by fredo** [comment]
@konradkonrad is this the same issue, you are experiencing

**Comment by konradkonrad** [comment]
Yes, that looks the same. I'm on current master.

**Comment by fredo** [comment]
This is the same problem as https://github.com/raiden-network/raiden/pull/6165

**Comment by palango** [comment]
Have you guys seen this? https://github.com/raiden-network/raiden/blob/develop/raiden/network/rpc/client.py#L419

Seems like there are some strange cases...

**Comment by fredo** [comment]
It depends on infura returning sometimes a valid response with the data property set to none at `getBlock`.

The attribute error comes from [geth_poa_middleware](https://github.com/fredo/raiden-wizard/blob/fcd9eeb9ee7eebd1c49707821ae7b3696a24ed8e/raiden_installer/ethereum_rpc.py#L49) (why are we actually using a poa middleware for mainnet?)

This was solved by creating an own middleware by @ulope and we should consider using the same.

**Comment by fredo** [comment]
should be fixed by #252
